libcontainer/configs/device.go

```diff
@@ -38,7 +38,13 @@ type Device struct {
 }
 
 func (d *Device) CgroupString() string {
-	return fmt.Sprintf("%c %s:%s %s", d.Type, deviceNumberString(d.Major), deviceNumberString(d.Minor), d.Permissions)
+	var p string
```

Comment:
you have to merge the permission when creating the deivces.  Think of the spec as the View and libcontainer config is the model.  You have to populate it correctly no matter how it is represented in the spec.